js/ui/map.js

```diff
@@ -212,14 +212,37 @@ class Map extends Camera {
     }
 
     /**
-     * Adds a [`Control`](#Control) to the map, calling `control.addTo(this)`.
+     * Adds a [`Control`](#Control) to the map, calling `control.onAdd(this)`.
      *
      * @param {Control} control The [`Control`](#Control) to add.
      * @returns {Map} `this`
      * @see [Display map navigation controls](https://www.mapbox.com/mapbox-gl-js/example/navigation/)
      */
     addControl(control) {
-        control.addTo(this);
+        control._map = this;
+        const container = control._container = control.onAdd(this);
```

Comment:
We give the control a reference to the map via both the `_map` property and the `onAdd` method. Is there a reason to give this reference twice? Could we standardize on one?